index.d.ts

```diff
@@ -0,0 +1,16 @@
+export default class AutocompleteElement extends HTMLElement {
+  src: string;
+  value: string;
+  open: boolean;
+}
+
+declare type AutocompleteEventType = 'auto-complete-change'
+
+declare type AutocompleteEvent$Init = CustomEvent & {
+  relatedTarget: HTMLInputElement;
+}
+
+export class AutocompleteEvent extends CustomEvent<any> {
```

Comment:
https://github.com/Microsoft/TypeScript/wiki/Breaking-Changes#customevent-is-now-a-generic-type

Reply:
`event.detail` should not be `any` in this case. 

```suggestion
export class AutocompleteEvent extends CustomEvent<null> {
```